src/foam/nanos/dig/DUGDAO.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.dig;
+
+import foam.core.Detachable;
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.AbstractSink;
+import foam.dao.DAO;
+import foam.dao.JDAO;
+import foam.dao.ProxyDAO;
+import foam.nanos.auth.User;
+
+public class DUGDAO
+    extends ProxyDAO
+{
+  public DUGDAO(X x) {
+    this(x, new JDAO(x, DUG.getOwnClassInfo(), "dugs"));
+  }
+
+  public DUGDAO(X x, DAO delegate) {
+    super(x, delegate);
+    delegate.select(new AbstractSink() {
+      @Override
+      public void put(FObject obj, Detachable sub) {
+        User user = (User) x.get("user");
+        ((DUG) obj).setOwner(user.getId());
```

Comment:
I think this line is a mistake. The user will be System at this point and we don't want to change the user to that. The user was already set in put_() when the object was originally created.

Reply:
The script needs to recreate the owner's context so that it doesn't run the DUG with extra permissions. To do this, see how the sudo() method is written in Script.js. Maybe even move that code somewhere reusable under auth.